File: generate_dark.py
import Cocoa

input_path = "macos/Runner/Assets.xcassets/AppIcon.appiconset/dark_1024.png"
output_path = "macos/Runner/Assets.xcassets/AppIcon.appiconset/dark_icon_base.png"

# Load the image
image = Cocoa.NSImage.alloc().initWithContentsOfFile_(input_path)

if not image:
    print("Failed to load image")
    exit(1)

size = image.size()
new_image = Cocoa.NSImage.alloc().initWithSize_(size)

new_image.lockFocus()

# Apple standard dark background (approx #1C1C1E)
Cocoa.NSColor.colorWithSRGBRed_green_blue_alpha_(28/255.0, 28/255.0, 30/255.0, 1.0).set()
Cocoa.NSRectFill(Cocoa.NSMakeRect(0, 0, size.width, size.height))

# Draw the transparent icon over it
image.drawInRect_(Cocoa.NSMakeRect(0, 0, size.width, size.height))

new_image.unlockFocus()

# Save out
cg_ref = new_image.CGImageForProposedRect_context_hints_(None, None, None)[0]
bitmap = Cocoa.NSBitmapImageRep.alloc().initWithCGImage_(cg_ref)
data = bitmap.representationUsingType_properties_(Cocoa.NSPNGFileType, None)
data.writeToFile_atomically_(output_path, True)

print("Dark base generated.")

File: generate_dark.swift
import Cocoa

let inputPath = "macos/Runner/Assets.xcassets/AppIcon.appiconset/dark_1024.png"
let outputPath = "macos/Runner/Assets.xcassets/AppIcon.appiconset/dark_icon_base.png"

guard let image = NSImage(contentsOfFile: inputPath) else {
    print("Failed to load image")
    exit(1)
}

let size = image.size
let newImage = NSImage(size: size)

newImage.lockFocus()

let color = NSColor(red: 28/255.0, green: 28/255.0, blue: 30/255.0, alpha: 1.0)
color.set()
let rect = NSRect(origin: .zero, size: size)
rect.fill()

image.draw(in: NSRect(origin: .zero, size: size))

newImage.unlockFocus()

guard let cgImage = newImage.cgImage(forProposedRect: nil, context: nil, hints: nil) else {
    exit(1)
}
let bitmap = NSBitmapImageRep(cgImage: cgImage)
if let data = bitmap.representation(using: .png, properties: [:]) {
    try? data.write(to: URL(fileURLWithPath: outputPath))
}
print("Dark base generated.")
